stock_split_picking_kit/wizards/stock_split_picking.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2025 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import fields, models
+from odoo.tools import groupby
+
+
+class StockSplitPicking(models.TransientModel):
+    _inherit = "stock.split.picking"
+
+    mode = fields.Selection(
+        selection_add=[("kit_quantity", "Quantity of kits")],
+        ondelete={"kit_quantity": "set default"},
+    )
+    kit_split_quantity = fields.Integer(string="Number of kits by transfer")
+    split_kit_order_move = fields.Char()
+
+    def _apply_kit_quantity(self):
+        pickings = self.env["stock.picking"]
+        for picking in self.mapped("picking_ids"):
+            pickings |= self._split_by_kit_quantity(picking)
+        return self._picking_action(pickings)
+
+    def _split_by_kit_quantity(self, picking):
+        filters = {
+            "incoming_moves": lambda m: m.location_id.usage == "supplier",
+            "outgoing_moves": lambda m: m.location_id.usage != "supplier",
```

Comment:
You group by picking, all moves are positive moves and should be considered as incoming_moves
```suggestion
            "incoming_moves": lambda m: True,
            "outgoing_moves": lambda m: False,
```